src/JitInterface/src/CorInfoImpl.cs

```diff
@@ -150,12 +176,19 @@ private IntPtr GetPin(Object obj)
             }
             return handle.AddrOfPinnedObject();
         }
-        private void FlushPins()
+
+        private List<IntPtr> _nativeExceptions = new List<IntPtr>();
```

Comment:
Updated the PR to create this list lazily, to remove lock around modifying it, and to set the list to null after using it.